src/main/java/edu/hawaii/its/api/controller/RestExceptionHandler.java

```diff
@@ -0,0 +1,98 @@
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+import java.net.URISyntaxException;
+import java.time.LocalDateTime;
+import java.util.List;
+
+@Order(Ordered.HIGHEST_PRECEDENCE)
+@ControllerAdvice
+public class RestExceptionHandler extends ResponseEntityExceptionHandler {
+    private ResponseEntity<Object> buildResponseEntity(ApiError apiError) {
+        return new ResponseEntity<>(apiError, apiError.getStatus());
+    }
+
+    // Other custom exception handlers using Builder pattern
+
+    @ExceptionHandler(ExceptionForTesting.class)
+    protected ResponseEntity<Object> handleExceptionForTesting(ExceptionForTesting ex) {
+
+        /* Common Error Message from @ExceptionForTesting.class */
+        /* status, message, timestamp must be nonNull */
+        ApiError.Builder errorBuilder = new ApiError.Builder()
+                .status(HttpStatus.INTERNAL_SERVER_ERROR)
+                .message("Validation failed")
+                .debugMessage("Check your service at endpoint /exception")
+                .timestamp(LocalDateTime.now());
+
+        /* GET SubErrors that defined from new exception object */
+        /* You can set the detail of SubErrors when you make throwable object */
+        List<ApiSubError> subErrors = ex.getSubErrors();
+
+        /* Add all SubErrors in the ApiError */
+        for (ApiSubError subError : subErrors) {
+            errorBuilder.addSubError(subError);
+        }
+
+        /* BUILD */
+        ApiError apiError = errorBuilder.build();
+
+        return buildResponseEntity(apiError);
+    }
+
+    @ExceptionHandler(IllegalArgumentException.class)
+    protected ResponseEntity<Object> handleIllegalArgumentException(IllegalArgumentException ex) {
+
+        ApiError.Builder errorBuilder = new ApiError.Builder()
+                .status(HttpStatus.FORBIDDEN)
+                .message("IllegalArgumentException")
+                .debugMessage("IllegalArgumentException debug message")
+                .timestamp(LocalDateTime.now());
+
+        /* BUILD */
+        ApiError apiError = errorBuilder.build();
+
+        return buildResponseEntity(apiError);
+    }
+
+    @ExceptionHandler(UhMemberNotFoundException.class)
+    protected ResponseEntity<Object> handleUhMemberNotFound(UhMemberNotFoundException ex) {
+
+        ApiError.Builder errorBuilder = new ApiError.Builder()
+                .status(HttpStatus.NOT_FOUND)
+                .message("UH Member found failed")
+                .debugMessage("This is not the validation error from frontend, check Sub Error to see the reason in the backend")
+                .timestamp(LocalDateTime.now());
+
+        List<ApiSubError> subErrors = ex.getSubErrors();
+
+        /* Add all SubErrors in the ApiError */
+        if(subErrors != null) {
+            for (ApiSubError subError : subErrors) {
+                errorBuilder.addSubError(subError);
+            }
+        }
```

Comment:
This null guard if statement is not needed, the for loop will not run in the subErrors List is empty.

Reply:
If the 86th line is absent, a 'NullPointerException' was occurring in an existing exception used within the service without the previous 'subError', so I made it.